docker-compose.yml

```
@@ -3,14 +3,14 @@ version: '3'
 services:
     scope:
         restart: always
-        image: bujesse/scope:1.0
+        build: .
         expose:
             - 55850
             - 55851
             - 55852
             - 55853
         volumes:
-            - static:/app
+            - ${SCOPE_DATA_DIR:-./scope_data/}:/root/.local/share/SCope/
         entrypoint: ["poetry", "run", "hypercorn", "--bind", "0.0.0.0", "main:scope_api"]
     httpd:
         restart: always
```

Would be nice to update the whole `docker-compose.yml` with this bits:
```suggestion
version: '3'

services:
    scope:
        restart: always
        build: .
        expose:
            - 55850
            - 55851
            - 55852
            - 55853
        volumes:
            - ${SCOPE_DATA_DIR:-./scope_data/}:/root/.local/share/SCope/
            - app-volume:/app
        entrypoint: ["poetry", "run", "hypercorn", "--bind", "0.0.0.0", "main:scope_api"]
    httpd:
        restart: always
        build: ./apache
        ports:
            - ${SCOPE_PORT:-80}:80
        depends_on:
            - scope
        volumes:
            - app-volume:/usr/local/apache2/htdocs/
volumes: 
    app-volume:
```

---

The problem with this is that you're serving the assets from inside the container (sharing volumes between apache and  the scope app instead of the host). Personally, I think it's better to serve them from outside the container so you can run "npm run build" from outside when you need to change config.json. This is useful when you need to change the publicHostAddress without rebuilding the image.

Is there a reason you want to serve assets from inside the container?

---

Indeed having to rebuild the image each time you change you config is annoying.
What I was trying to avoid is having the `npm run build` run on different Node version than what the Docker image is using (v14).
I have Node `v9` on my computer. I followed the steps you wrote in the `README.md` and I wasn't not able to start SCope with the `docker-compose.yml` file.
I think this is because discrepancies between 2 node versions. This is why I suggested to run `npm run build` within the Docker container.
Maybe instead we could leave like that but update the `README.md` with and build the assets with the scope container ? What do you think ?

---

Oh I see, I could see that being a problem. 

Are you thinking we should build the assets from within the container, but then have a shared volume with the host, so it still gets served from the host in the end? Or were you thinking to share a volume between scope and apache and have the assets built and served inside, like you had suggested?

---

Yeah the former. I think your point of serving the assets from the host is a strong point so let's keep that.
But then instead of building the assets with the host, we could use the SCope docker instead in order to keep Node version consistent ? (So the `README.md` should be updated then)

---

Got it, I'll make the change and test it

---

Apparently you cannot copy files from a container in the Dockerfile, but you _can_ do it afterwards manually with the "docker cp" command. Would it be alright for me to add to the README that if you need to, you can grab the assets from within the container using that command?

---

Or possibly even better; have 2 docker-compose files, one for serving from host, and one serving from container.

---

Yes I think it's fine

---

Please see the new changes and try again

---

Just tested and it works for me when I follow the example to serve from container :+1: